source/FreeRTOS_IPv6.c

```diff
@@ -476,10 +494,9 @@ eFrameProcessingResult_t prvAllowIPPacketIPv6( const IPHeader_IPv6_t * const pxI
             eReturn = eProcessBuffer;
         }
         /* Is it the legal multicast address? */
-        else if( ( xHasUnspecifiedAddress == pdFALSE ) &&
+        else if( ( ( xHasUnspecifiedAddress == pdFALSE ) &&
+                   ( xBadIPv6Loopback( pxIPv6Header ) == pdFALSE ) ) &&
```

Comment:
Codewise it makes more sense to have xBadIPv6Loopback and xBadIPv4Loopback checks at the same level. Here, xBadIPv4Loopback check is happening at FreeRTOS_IP.c but xBadIPv6Loopback check is in IPv6 specfic file. Can we re-arrange the code.

Reply:
I answered this and rearranged the code like this:

`prvAllowIPPacketIPv4()` calls `xBadIPv4Loopback()`
`prvAllowIPPacketIPv6()` calls `xBadIPv6Loopback()`